shamir/README.md

```diff
@@ -0,0 +1 @@
+Forked from [Vault](https://github.com/hashicorp/vault/tree/master/shamir)
```

Comment:
also put a copy of the original LICENSE file in this directory, even if it's the same as the one `sops` uses.